### PR TITLE
Add the incoming mapper to the integration instance

### PR DESCRIPTION
**Pull request by franciscosd**
Whilst we considered adding variables like the incoming mapper id in the config map, we came to the conclusion that is was more logical to add it as a separate variable inside the resource.

In terms of testing of the code, we tested the creating, updating and destruction of integration instance resources with and without the incoming mapper id.

Good thing to be aware of, is that XSOAR lacks input validation in the API. This means that you can provide invalid incoming mapper id's, but you can also provide fields that XSOAR says it requires - whilst it does not return errors if you don't provide it. We've already started a conversation about this with our Palo Alto contacts, as we think this should be fixed in XSOAR - and we should not be required to add such validation into the Terraform provider.

### CODE REVIEW

File: xsoar/resource_integration_instance.go [hunk 1]
```diff
@@ -459,6 +466,7 @@ func (r resourceIntegrationInstance) ImportState(ctx context.Context, req tfsdk.
 		IntegrationName:   types.String{Value: integration["brand"].(string)},
 		PropagationLabels: types.List{Elems: propagationLabels, ElemType: types.StringType},
 		Config:            types.Map{},
+		IncomingMapperId:  types.String{},
```
Comment (jrauen) [inline xsoar/resource_integration_instance.go:469]: This value should be available from the API response

File: xsoar/data_source_integration_instance.go [hunk 2]
```diff
@@ -112,7 +113,7 @@ func (r dataSourceIntegrationInstance) Read(ctx context.Context, req tfsdk.ReadD
 		Account:           config.Account,
 		PropagationLabels: types.List{Elems: propagationLabels, ElemType: types.StringType},
 		Config:            config.Config,
-		IncomingMapperId:  config.IncomingMapperId,
+		IncomingMapperId:  types.String{Value: integration["incomingMapperId"].(string)},
```
Comment (badarsebard) [inline xsoar/data_source_integration_instance.go:116]: This has the potential to panic if the value of the `interface{}` is `nil`. When using type conversion in `nil` it's suggested to use multiple assignment and verify the result.

```go
incomingMapperId, ok := integration["incomingMapperId"].(string)
if ok {
    result.incomingMapperId = types.String{Value: incomingMapperId}
} else {
    result.incomingMapperId = types.String{Null: true}
}
```

File: xsoar/resource_integration_instance.go [hunk 3]
```diff
@@ -184,7 +185,7 @@ func (r resourceIntegrationInstance) Create(ctx context.Context, req tfsdk.Creat
 		Account:           plan.Account,
 		PropagationLabels: types.List{Elems: propagationLabels, ElemType: types.StringType},
 		Config:            plan.Config,
-		IncomingMapperId:  plan.IncomingMapperId,
+		IncomingMapperId:  types.String{Value: integration["incomingMapperId"].(string)},
```
Comment (badarsebard) [inline xsoar/resource_integration_instance.go:188]: Same as above

File: xsoar/resource_integration_instance.go [hunk 4]
```diff
@@ -246,7 +247,7 @@ func (r resourceIntegrationInstance) Read(ctx context.Context, req tfsdk.ReadRes
 		Account:           state.Account,
 		PropagationLabels: types.List{Elems: propagationLabels, ElemType: types.StringType},
 		Config:            state.Config,
-		IncomingMapperId:  state.IncomingMapperId,
+		IncomingMapperId:  types.String{Value: integration["incomingMapperId"].(string)},
```
Comment (badarsebard) [inline xsoar/resource_integration_instance.go:250]: Same as above

File: xsoar/resource_integration_instance.go [hunk 5]
```diff
@@ -378,7 +379,7 @@ func (r resourceIntegrationInstance) Update(ctx context.Context, req tfsdk.Updat
 		Account:           plan.Account,
 		PropagationLabels: types.List{Elems: propagationLabels, ElemType: types.StringType},
 		Config:            plan.Config,
-		IncomingMapperId:  plan.IncomingMapperId,
+		IncomingMapperId:  types.String{Value: integration["incomingMapperId"].(string)},
```
Comment (badarsebard) [inline xsoar/resource_integration_instance.go:382]: Same as above

File: xsoar/resource_integration_instance.go [hunk 6]
```diff
@@ -466,7 +467,7 @@ func (r resourceIntegrationInstance) ImportState(ctx context.Context, req tfsdk.
 		IntegrationName:   types.String{Value: integration["brand"].(string)},
 		PropagationLabels: types.List{Elems: propagationLabels, ElemType: types.StringType},
 		Config:            types.Map{},
-		IncomingMapperId:  types.String{},
+		IncomingMapperId:  types.String{Value: integration["incomingMapperId"].(string)},
```
Comment (badarsebard) [inline xsoar/resource_integration_instance.go:470]: Same as above

**Review by badarsebard — CHANGES_REQUESTED** [review]

**Comment by fancybear-dev** [comment]
The incoming mapper id should be OK now. Have tested it quite extensively, and had no issues.

**Review by jrauen — APPROVED** [review]

**Pull request merged**